### PR TITLE
Added standalone action handling

### PR DESCRIPTION
**Pull request by Robert-SD**

### CODE REVIEW

File: android/src/main/kotlin/com/adyen/checkout/flutter/PlatformApi.kt
Comment (Robert-SD) [inline android/src/main/kotlin/com/adyen/checkout/flutter/PlatformApi.kt:1]: Generated code - no review required.

File: ios/Classes/PlatformApi.swift
Comment (Robert-SD) [inline ios/Classes/PlatformApi.swift:1]: Generated code - no review required.

File: lib/src/generated/platform_api.g.dart
Comment (Robert-SD) [inline lib/src/generated/platform_api.g.dart:1]: Generated code - no review required.

File: android/src/main/kotlin/com/adyen/checkout/flutter/components/action/ActionComponentCallback.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import ComponentFlutterInterface
+import PaymentResultDTO
+import PaymentResultEnum
+import androidx.fragment.app.FragmentActivity
+import com.adyen.checkout.components.core.ActionComponentCallback
+import com.adyen.checkout.components.core.ActionComponentData
+import com.adyen.checkout.components.core.ComponentError
+import com.adyen.checkout.flutter.components.view.ComponentLoadingBottomSheet
+
+class ActionComponentCallback(
```
Comment (OscarSpruit) [inline android/src/main/kotlin/com/adyen/checkout/flutter/components/action/ActionComponentCallback.kt:13]: Could this be `internal`?

File: android/src/main/kotlin/com/adyen/checkout/flutter/components/action/ActionComponentManager.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import androidx.fragment.app.FragmentActivity
+import com.adyen.checkout.action.core.GenericActionComponent
+import com.adyen.checkout.components.core.ActionComponentCallback
+import com.adyen.checkout.components.core.action.Action
+import com.adyen.checkout.flutter.components.view.ComponentLoadingBottomSheet
+import com.adyen.checkout.flutter.utils.ConfigurationMapper.mapToCheckoutConfiguration
+import org.json.JSONObject
+import java.util.UUID
+
+class ActionComponentManager(
```
Comment (OscarSpruit) [inline android/src/main/kotlin/com/adyen/checkout/flutter/components/action/ActionComponentManager.kt:14]: and can this one also be `internal?`

File: android/src/main/kotlin/com/adyen/checkout/flutter/components/action/ActionComponentManager.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        val genericActionComponent =
+            GenericActionComponent.PROVIDER.get(
+                activity,
+                checkoutConfiguration,
+                ActionComponentCallback(
+                    activity,
+                    componentFlutterApi,
+                    componentId,
+                ),
+                UUID.randomUUID().toString()
```
Comment (OscarSpruit) [inline android/src/main/kotlin/com/adyen/checkout/flutter/components/action/ActionComponentManager.kt:35]: Will this cause a new component to be created when you rotate the screen for example?

File: android/src/main/kotlin/com/adyen/checkout/flutter/utils/ConfigurationMapper.kt [hunk 4]
```diff
@@ -375,6 +376,16 @@ object ConfigurationMapper {
         }
     }
 
+    fun InstantPaymentConfigurationDTO.mapToGooglePayCheckoutConfiguration(amount: Amount): CheckoutConfiguration {
+        return CheckoutConfiguration(
+            environment.mapToEnvironment(),
+            clientKey,
+            shopperLocale?.let { Locale.forLanguageTag(it) },
```
Comment (OscarSpruit) [inline android/src/main/kotlin/com/adyen/checkout/flutter/utils/ConfigurationMapper.kt:383]: Does `Locale.forLanguageTag` throw an exception when the language tag is invalid?

File: ios/Classes/components/ComponentPlatformApi.swift [hunk 5]
```diff
@@ -61,12 +63,20 @@ class ComponentPlatformApi: ComponentPlatformInterface {
             )
         }
     }
+    
+    func handleAction(actionComponentConfiguration: ActionComponentConfigurationDTO, componentId: String, actionResponse: [String?: Any?]?) throws {
+        guard let actionResponse else { return }
```
Comment (goergisn) [inline ios/Classes/components/ComponentPlatformApi.swift:68]: Is `[String?: Any?]?` really needed - can't you just pass an empty hashmap?

**Review by goergisn — APPROVED** [review]

**Review by goergisn — APPROVED** [review]

**Review by OscarSpruit — APPROVED** [review]

**Pull request merged**